In [5]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
import math

## Basic exploration of data

In [6]:
### Note: Area of houses are given in square feet, prices are given in US$
df_homeprice = pd.read_excel('home_price3.xlsx')
df_homeprice.head(10)

Area  bedroom  age   price
0  2600      3.0   20  550000
1  3000      4.0   15  565000
2  3200      NaN   18  610000
3  3600      3.0   30  595000
4  4000      5.0    8  760000

## Cleaning the data

In [7]:
median_bedrooms = math.floor(df_homeprice['bedroom'].median())
median_bedrooms.real

3

In [8]:
x = df_homeprice['bedroom'].fillna(median_bedrooms)

In [9]:
df_homeprice['bedroom'] = x

In [10]:
# Areas of houses are in square feet, prices are in US$
df_homeprice.head(10)

Area  bedroom  age   price
0  2600      3.0   20  550000
1  3000      4.0   15  565000
2  3200      3.0   18  610000
3  3600      3.0   30  595000
4  4000      5.0    8  760000

## Making linearregressionmodel

In [13]:
reg = LinearRegression()
reg.fit(df_homeprice[['Area','bedroom','age']], df_homeprice['price'])

LinearRegression()

In [14]:
### Coefficients of independent variables ['Area', 'bedroom', 'age']
reg.coef_

array([   137.25, -26025.  ,  -6825.  ])

In [15]:
### Value of intercept
reg.intercept_

383725.00000000006

## AIM: To find the price of a house with Area = 3000 sq. ft. ,3 bedrooms, age=40 years

In [16]:
reg.predict([[3000,3,40]])

c:\Python\Python37\lib\site-packages\sklearn\base.py:451: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  "X does not have valid feature names, but"


array([444400.])

##### RESULT: The price of the house is approx. US$ 444400

## AIM: To predict the price of house of Area 2500 sq ft, 4 bedrooms, age 5 yrs old

In [17]:
reg.predict([[2500,4,5]])

c:\Python\Python37\lib\site-packages\sklearn\base.py:451: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  "X does not have valid feature names, but"


array([588625.])

##### RESULT: The price of house is approx. US$ 588625